packages/edit-post/src/components/layout/index.js

```diff
@@ -250,19 +331,59 @@ function MetaBoxesMain( { isLegacy } ) {
 				target.setPointerCapture( pointerId );
 			} }
 			onResizeStart={ ( event, direction, elementRef ) => {
-				// Avoids height jumping in case it’s limited by max-height.
-				elementRef.style.height = `${ elementRef.offsetHeight }px`;
-				// Stops initial max-height from being applied.
-				elementRef.classList.add( 'has-user-size' );
+				if ( isAutoHeight ) {
+					const heightNow = elementRef.offsetHeight;
+					// Sets the starting height to avoid visual jumps in height and
+					// aria-valuenow being `NaN` for the first (few) resize events.
+					resizableBoxRef.current.updateSize( { height: heightNow } );
+					// Causes `maxHeight` to update to full `max` value instead of half.
+					setIsUntouched( false );
+				}
+			} }
+			onResize={ ( event, direction, elementRef, delta ) => {
+				const fromHeight = resizableBoxRef.current.state.height;
+				const nextHeight = fromHeight + delta.height;
+				const separator = separatorRef.current;
+				separator.ariaValueNow = getAriaValueNow( nextHeight, max );
 			} }
 			onResizeStop={ () => {
+				const nextHeight = resizableBoxRef.current.state.height;
 				setPreference(
 					'core/edit-post',
 					'metaBoxesMainOpenHeight',
-					resizableBoxRef.current.state.height
+					nextHeight
 				);
 			} }
+			handleClasses={ {
+				top: 'edit-post-meta-boxes-main__resize-handle',
+			} }
+			handleComponent={ {
+				top: (
+					<>
+						<Tooltip text={ __( 'Drag to resize' ) }>
+							{ /* Disable reason: aria-valuenow is supported by separator role. */ }
+							{ /* eslint-disable-next-line jsx-a11y/role-supports-aria-props */ }
+							<button
+								ref={ separatorRef }
+								aria-label={ __( 'Drag to resize' ) }
```

Comment:
```suggestion
								aria-orientation="horizontal"
								aria-label={ __( 'Drag to resize' ) }
```

It looks like the site editor canvas has [this aria attribute defined](https://github.com/WordPress/gutenberg/blob/5ddf1c376beaf1c1976ecc5fc0a5200cc1eb9068/packages/edit-site/src/components/resizable-frame/index.js#L280), what do you think?

Reply:
Thanks for asking. I had read that ["default is horizontal for separator"](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/separator_role#associated_wai-aria_roles_states_and_properties) so figured it could be left out. Also, it seems easy to confuse which value is correct but I think this is.